pkgs/pkgs-lib/formats.nix

```diff
@@ -48,13 +48,19 @@ rec {
 
   };
 
-  # YAML has been a strict superset of JSON since 1.2
   yaml = {}:
     let jsonSet = json {};
     in jsonSet // {
```

Comment:
This should no longer inherit from `json`.

Reply:
> This should no longer inherit from `json`.

The typing validation works in the same way for both, that was right. The proposition is to change how the YAML is serialized.

Reply:
The issue with inheriting attributes from `json` is that it's not future proof.
We can not assume that future attributes from `json` apply to `yaml` too.

Reply:
By specification json is a subset of yaml. That way was right according to the specification. This PR is just to generate a YAMLish syntax. JSON as YAML works, software understand it but sometimes you want to generate dotfiles and its a bit harder to grasp what is going on compared to the documentation of the upstream software.

Reply:
And YAML as JSON does not work, so we should not inherit attributes from json unless we know that such attributes do not cause code that requires JSON to accept the YAML type. This is why inheriting from `json` is fragile.